File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dum=ListNode(0)
        i=head
        p=dum
        dum.next=head
        
        while i:
            if i.next and i.next.val==i.val:
                while i.next and i.next.val==i.val:
                    i.next=i.next.next
                p.next=i.next
            else:
                p=p.next
            i=i.next

        return dum.next

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *i=head;
        ListNode *d=new ListNode(0);
        ListNode *p=d;
        
        d->next=head;
        while(i){
            if(i->next && i->val==i->next->val){
                while(i->next && i->val==i->next->val){
                    i=i->next;
                }
                p->next=i->next;
            }else{
                p=p->next;                
            }           
            i=i->next;
        }
        return d->next;
    }
};